src/components/DailyRoutinePage/AddedProductCard.js

```diff
@@ -2,23 +2,59 @@ import "./AddedProductCard.css";
 import { TiShoppingCart } from "react-icons/ti";
 import bottleimg from "../../images/niacinamide-bottle.png";
 
-export default function AddedProductCard() {
+import getProductsById from "../../utility/getProductsById";
+
+export default function AddedProductCard({ info, products, name, time }) {
+  let i;
+  if (name === "Monday") {
+    i = 0;
+  } else if (name === "Tuesday") {
+    i = 1;
+  } else if (name === "Wednesday") {
+    i = 2;
+  } else if (name === "Thursday") {
+    i = 3;
+  } else if (name === "Friday") {
+    i = 4;
+  } else if (name === "Saturday") {
+    i = 5;
+  } else if (name === "Sunday") {
+    i = 6;
+  }
+
+  const weekdayTime = info.filter((product) => {
+    return product.days[i][time];
+  });
+
+  const openingDate = info.filter((product) => {
+    return product.date;
+  });
+
+  console.log(products);
+
   return (
-    <div className="AddedProductCard">
-      <h5 className="added-product-name">Niacinamide 10% + Zinc 1%</h5>
-      <div className="time-details">
-        <p>opened: february</p>
-        <p>lasts: 12 months</p>
-      </div>
-      <p className="restock">
-        running low? <TiShoppingCart className="shopping-cart-icon" />{" "}
-      </p>
-      <button className="edit-button">edit</button>
-      <img
-        src={bottleimg}
-        alt="product bottle"
-        className="added-bottle-image"
-      ></img>
-    </div>
+    products.length > 0 &&
+    weekdayTime.map((time) => {
+      return (
+        <div className="AddedProductCard">
+          <h5 className="added-product-name">
+            {getProductsById(time.id, products).name}
```

Comment:
You could just call `getProductsById` once before the `return` inside your `.map` function, store the result into a variable and then you don't need to call it anymore, just use that variable.

Reply:
hecho ;)